cai_causal_graph/causal_graph.py

```diff
@@ -1726,6 +1726,49 @@ def get_common_descendants(self, node_1: NodeLike, node_2: NodeLike) -> Set[str]
 
         return self.get_descendants(node_1).intersection(self.get_descendants(node_2))
 
+    def get_nodes_between(self, node_1: NodeLike, node_2: NodeLike) -> Set[Node]:
+        """
+        Get the set of nodes that are on causal paths between two nodes.
+
+        The nodes themselves will be included in the final set.
+
+        If there are no causal paths between the nodes then an empty set will be returned.
+
+        :param node_1: a single node-identifier coercible object, representing the source node.
+        :param node_2: a single node-identifier coercible object, representing the destination node.
+        :return: a set of nodes that are on causal paths between the two provided nodes.
+        """
+
+        # Set up cache: for each seen node, cache whether there is a causal path between it and the destination
+        seen_nodes: dict[str, bool] = {}
+
+        start = self.get_node(node_1)
+        end = self.get_node(node_2)
+
+        def _get_nodes_between_inner(start_node_: Node, end_node_: Node) -> bool:
+            if start_node_.identifier in seen_nodes:
+                return seen_nodes[start_node_.identifier]
+            if start_node_ == end_node_:
+                seen_nodes[start_node_.identifier] = True
+                return True
+            if start_node_.is_sink_node():
+                seen_nodes[start_node_.identifier] = False
+                return False
+
+            start_children = [e.destination for e in start_node_._outbound_edges]
+            has_causal_path = any([_get_nodes_between_inner(child, end_node_) for child in start_children])
```

Comment:
could you add a comment above, as this line can get buggy:

if you omitted the explicit list construction, and instead ran `any` directly on the list comprehension, this would produce buggy behaviour, because `any` would run until the first `True` and return.

Could you just add a comment above saying the explicit list construction is needed. Just for future maintenance etc. I could see someone going through this code and thinking "oh, we could make this more efficient" and then introducing this bug.